src/main/java/org/elasticsearch/index/query/QueryParsingException.java

```diff
@@ -19,21 +19,52 @@
 
 package org.elasticsearch.index.query;
 
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentLocation;
 import org.elasticsearch.index.Index;
 import org.elasticsearch.index.IndexException;
 import org.elasticsearch.rest.RestStatus;
 
+import java.io.IOException;
+
 /**
  *
  */
 public class QueryParsingException extends IndexException {
 
-    public QueryParsingException(Index index, String msg) {
-        super(index, msg);
+    public static final int UNKNOWN_POSITION = -1;
+    private int lineNumber = UNKNOWN_POSITION;
+    private int columnNumber = UNKNOWN_POSITION;
+
+    public QueryParsingException(Index index, String msg, @Nullable XContentLocation location) {
```

Comment:
>maybe have a ctor without the location since it can be null?

I wanted to enforce the new behaviour of passing location info from devs writing parsers. The vast majority of the existing parser errors are able to pass this info. I can remove the requirement if you want. It was certainly useful to use the compiler to ensure I had covered all the parsers while making this PR.